control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -222,19 +237,24 @@ func (r *HostedControlPlaneReconciler) Reconcile(ctx context.Context, req ctrl.R
 	originalHostedControlPlane := hostedControlPlane.DeepCopy()
 	// This is a best effort ping to the identity provider
 	// that enables access from the operator to the cloud provider resources.
-	healthCheckIdentityProvider(ctx, hostedControlPlane)
+	healthCheckIdentityProvider(ctx, hostedControlPlane, r.ec2Client)
 
 	// Return early if deleted
 	if !hostedControlPlane.DeletionTimestamp.IsZero() {
+		if err := r.destroyAWSDefaultSecurityGroup(ctx, hostedControlPlane); err != nil {
```

Comment:
Should this be inside/after `shouldCleanupCloudResources`? The security group will not be able to be deleted if any worker EC2 instances remain

Reply:
By the time the hcp is deleted, all instances have been terminated. I didn't want to use the `shouldCleanupCloudResources` function because it also checks whether there is an annotation to cleanup resources on the HostedCluster. We want to create/destroy the SG regardless. However, both end up checking that we still have valid STS creds.